source/grid/grid_refinement.cc

```diff
@@ -515,26 +466,29 @@ GridRefinement::refine_and_coarsen_optimize (Triangulation<dim,spacedim> &tria,
           ExcDimensionMismatch(criteria.size(), tria.n_active_cells()));
   Assert (criteria.is_non_negative (), ExcNegativeCriteria());
 
-  // get an increasing order on
-  // the error indicator
-  std::vector<unsigned int> tmp(criteria.size());
-  for (unsigned int i=0; i<criteria.size(); ++i)
-    tmp[i] = i;
+  // get a decreasing order on the error indicator
+  std::vector<types::global_dof_index> cell_indices(criteria.size());
```

Comment:
That seems unnecessary. We're not likely ever going to be able to store more than 2B cells on a single processor -- or even if we can store them, it's not useful to do this many.